network/p2p/middleware.go

```diff
@@ -32,17 +32,30 @@ const (
 )
 
 const (
+	_ = iota
+	_ = 1 << (10 * iota)
+	mb
+	gb
+)
+
+const (
+
 	// defines maximum message size in publish and multicast modes
-	DefaultMaxPubSubMsgSize = 1 << 21 // 2 mb
+	DefaultMaxPubSubMsgSize = 5 * mb // 5 mb
+
+	// defines maximum message size in unicast mode for most messages
+	DefaultMaxUnicastMsgSize = 10 * mb // 10 mb
 
-	// defines maximum message size in unicast mode
-	DefaultMaxUnicastMsgSize = 5 * DefaultMaxPubSubMsgSize // 10 mb
+	// defines maximum message size in unicast mode for large messages
+	LargeMsgMaxUnicastMsgSize = gb // 1 gb
 
-	// maximum time to wait for a unicast request to complete
+	// default maximum time to wait for a default unicast request to complete
+	// assuming at least a 1mb/sec connection
 	DefaultUnicastTimeout = 2 * time.Second
-)
 
-var unicastTimeout = DefaultUnicastTimeout
+	// maximum time to wait for a unicast request to complete for large message size
+	LargeMsgUnicastTimeout = 1000 * time.Second
```

Comment:
the `1gb` large message will take more time to transfer and hence a bigger unicast timeout. The time out is based on the previous timeout of 2 second for a 2 mb message (1mb/sec).